File: spring-boot/src/main/java/com/magroun/gestiondesfactures/service/InvoiceService.java
package com.magroun.gestiondesfactures.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import com.magroun.gestiondesfactures.model.Invoice;

public interface InvoiceService {
    Invoice createInvoice(Invoice invoice);
    Page<Invoice> getAllInvoices(Pageable pageable);
    Page<Invoice> getAllQuotes(Pageable pageable);
    Invoice getInvoiceById(Long id);
    Invoice updateInvoice(Long id, Invoice updatedInvoice);
    void deleteInvoice(Long id);
}

File: angular-16/src/app/services/invoice.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Invoice } from '../models/invoice';
import { Page } from '../models/page';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class InvoiceService {
  private readonly apiUrl = environment.apiUrl.invoices;

  constructor(private http: HttpClient) { }

  getAllInvoices(page: number, size: number): Observable<Page<any>> {
    let params = new HttpParams()
      .set('page', page.toString())
      .set('size', size.toString());

    return this.http.get<Page<any>>(`${this.apiUrl}`,{ params });
  }

  getAllQuotes(page: number, size: number): Observable<Page<any>> {
    let params = new HttpParams()
      .set('page', page.toString())
      .set('size', size.toString());

    return this.http.get<Page<any>>(`${this.apiUrl}/quotes`,{ params });
  }

  getInvoiceById(id: number): Observable<Invoice> {
    return this.http.get<Invoice>(`${this.apiUrl}/${id}`);
  }

  createInvoice(invoice: Invoice): Observable<Invoice> {
    return this.http.post<Invoice>(`${this.apiUrl}`, invoice);
  }

  updateInvoice(id: number, invoice: Invoice): Observable<Invoice> {
    return this.http.put<Invoice>(`${this.apiUrl}/${id}`, invoice);
  }

  deleteInvoice(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
